lib/cn/wbt_reader.c

```diff
@@ -903,6 +903,38 @@ wbti_fini(void)
     wbti_cache = NULL;
 }
 
+static bool HSE_NONNULL(1)
+wbtr_hdr_valid(const struct wbt_hdr_omf *omf)
+{
+    const uint32_t version = omf_wbt_version(omf);
+    const uint32_t magic = omf_wbt_magic(omf);
+
+    return HSE_LIKELY(version == WBT_TREE_VERSION && magic == WBT_TREE_MAGIC);
+}
+
+merr_t
+wbtr_read_desc(const struct wbt_hdr_omf *wbt_hdr, struct wbt_desc *desc)
+{
+    if (!wbtr_hdr_valid(wbt_hdr))
+        return merr(ev(EINVAL));
+
+    desc->wbd_version = omf_wbt_version(wbt_hdr);
+
+    switch (desc->wbd_version) {
+        case WBT_TREE_VERSION:
+            desc->wbd_root = omf_wbt_root(wbt_hdr);
+            desc->wbd_leaf = omf_wbt_leaf(wbt_hdr);
+            desc->wbd_leaf_cnt = omf_wbt_leaf_cnt(wbt_hdr);
+            desc->wbd_kmd_pgc = omf_wbt_kmd_pgc(wbt_hdr);
+            break;
+
+        default:
+            abort();
+    }
```

Comment:
Align case labels with switch to prevent the illusion of a block scope that doesn't exist...